src/SFA.DAS.EmployerAccounts/Queries/GetApprenticeship/GetApprenticeshipsHandler.cs

```diff
@@ -35,12 +38,39 @@ public async Task<GetApprenticeshipsResponse> Handle(GetApprenticeshipsRequest m
                 throw new InvalidRequestException(validationResult.ValidationDictionary);
             }
 
-            long accountId = _hashingService.DecodeValue(message.HashedAccountId);
-
-            return new GetApprenticeshipsResponse
+            try
+            {
+                long accountId = _hashingService.DecodeValue(message.HashedAccountId);
+                var task = _commitmentV2Service.GetApprenticeships(accountId);
+                if (await Task.WhenAny(task, Task.Delay(_employerAccountsConfiguration.AddApprenticeCallToActionTimeout)) == task)
```

Comment:
This functionality could be added using the decorator pattern.  That would keep this code unchanged and follow the single responsibility principle.  Also as we have used Polly elsewhere in the system it would be better to use that to implement the timeout since it is well tested.